data-structures/ruby/doubly_linked_list/base.rb

```diff
@@ -0,0 +1,123 @@
+require_relative 'empty'
+require_relative 'node'
+require_relative 'doubly_linked_list_node'
+
+module DoublyLinkedList
+  class Base
+
+    include DoublyLinkedList
+    include Enumerable
+
+    attr_reader :head, :tail, :length
+    def initialize()
+      empty = DoublyLinkedList::Empty.new()
+      @head = empty
+      @tail = empty
+      @length = 0
+    end
+
+    def tail
+      @tail
+    end
+
+    def calculate_length
+      length = 0
+      placeholder = @head
+      until placeholder.is_a?(DoublyLinkedList::Empty) do
+        length += 1
+        placeholder = placeholder.next
+      end
+      length
+    end
+
+    def find_tail
```

Comment:
What is the point of this method?  Aren't you keeping track of the tail directly?